### ISSUE TITLE
[Vite] `Cannot find package 'react-refresh' imported from ... babel-virtual-resolve-base.js` when running custom server with pnpm during dev

### ISSUE THREAD

**Issue by hi-ogawa** [remix-run/remix#7905]
### What version of Remix are you using?

2.2.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

```sh
$ pnpm --version
8.10.0

# latest = 2.2.0 at this time
$ pnpm dlx create-remix@latest --template remix-run/remix/templates/unstable-vite-express remix-vite-express-pnpm
...

$ cd remix-vite-express-pnpm
$ pnpm dev  # then access http://localhost:3000/

> remix-vite-express-pnpm@ dev /home/hiroshi/code/tmp/remix-vite-express-pnpm
> node ./server.mjs

http://localhost:3000
오후 4:34:45 [vite] Internal server error: Cannot find package 'react-refresh' imported from /home/hiroshi/code/tmp/remix-vite-express-pnpm/babel-virtual-resolve-base.js
  Plugin: remix-react-refresh-babel
...(see below for full stacktrace)...
```

<details><summary>reveal full log</summary>

```
$ pnpm dev

> remix-vite-express-pnpm@ dev /home/hiroshi/code/tmp/remix-vite-express-pnpm
> node ./server.mjs

http://localhost:3000
오후 4:34:45 [vite] Internal server error: Cannot find package 'react-refresh' imported from /home/hiroshi/code/tmp/remix-vite-express-pnpm/babel-virtual-resolve-base.js
  Plugin: remix-react-refresh-babel
  File: /home/hiroshi/code/tmp/remix-vite-express-pnpm/app/root.tsx
      at new NodeError (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/vendor/import-meta-resolve.js:203:5)
      at packageResolve (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/vendor/import-meta-resolve.js:873:9)
      at moduleResolve (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/vendor/import-meta-resolve.js:902:20)
      at defaultResolve (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/vendor/import-meta-resolve.js:985:15)
      at resolve (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/vendor/import-meta-resolve.js:999:12)
      at tryImportMetaResolve (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/config/files/plugins.js:137:45)
      at resolveStandardizedNameForImport (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/config/files/plugins.js:159:19)
      at resolveStandardizedName (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/config/files/plugins.js:168:12)
      at loadPlugin (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/config/files/plugins.js:47:20)
      at loadPlugin.next (<anonymous>)
      at createDescriptor (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/config/config-descriptors.js:140:16)
      at createDescriptor.next (<anonymous>)
      at step (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/gensync@1.0.0-beta.2/node_modules/gensync/index.js:261:32)
      at evaluateAsync (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/gensync@1.0.0-beta.2/node_modules/gensync/index.js:291:5)
      at /home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/gensync@1.0.0-beta.2/node_modules/gensync/index.js:44:11
      at Array.forEach (<anonymous>)
      at Function.async (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/gensync@1.0.0-beta.2/node_modules/gensync/index.js:43:15)
      at Function.all (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/gensync@1.0.0-beta.2/node_modules/gensync/index.js:216:13)
      at Generator.next (<anonymous>)
      at createDescriptors (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/config/config-descriptors.js:102:41)
      at createDescriptors.next (<anonymous>)
      at createPluginDescriptors (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/config/config-descriptors.js:99:17)
      at createPluginDescriptors.next (<anonymous>)
      at /home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/config/config-descriptors.js:65:32
      at Generator.next (<anonymous>)
      at Function.<anonymous> (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/gensync-utils/async.js:21:3)
      at Generator.next (<anonymous>)
      at step (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/gensync@1.0.0-beta.2/node_modules/gensync/index.js:269:25)
      at evaluateAsync (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/gensync@1.0.0-beta.2/node_modules/gensync/index.js:291:5)
      at Function.errback (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/gensync@1.0.0-beta.2/node_modules/gensync/index.js:113:7)
      at errback (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/gensync-utils/async.js:66:18)
      at async (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/gensync@1.0.0-beta.2/node_modules/gensync/index.js:188:17)
      at onFirstPause (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/gensync@1.0.0-beta.2/node_modules/gensync/index.js:216:13)
      at Generator.next (<anonymous>)
      at cachedFunction (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/config/caching.js:52:46)
      at cachedFunction.next (<anonymous>)
      at mergeChainOpts (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/config/config-chain.js:350:34)
      at mergeChainOpts.next (<anonymous>)
      at chainWalker (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/config/config-chain.js:317:14)
      at chainWalker.next (<anonymous>)
      at buildRootChain (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/config/config-chain.js:57:36)
      at buildRootChain.next (<anonymous>)
      at loadPrivatePartialConfig (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/config/partial.js:72:62)
      at loadPrivatePartialConfig.next (<anonymous>)
      at loadFullConfig (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/config/full.js:36:46)
      at loadFullConfig.next (<anonymous>)
      at transform (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/@babel+core@7.23.2/node_modules/@babel/core/lib/transform.js:20:44)
      at transform.next (<anonymous>)
      at step (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/gensync@1.0.0-beta.2/node_modules/gensync/index.js:269:25)
      at evaluateAsync (/home/hiroshi/code/tmp/remix-vite-express-pnpm/node_modules/.pnpm/gensync@1.0.0-beta.2/node_modules/gensync/index.js:291:5)
```

</details>

<details><summary>reveal screenshot</summary>

![image](https://github.com/remix-run/remix/assets/4232207/b575ff47-3103-4cff-9ffb-9161acc688cd)

</details>

### Expected Behavior

Expected to work same as npm

### Actual Behavior

Error as written in "Steps to Reproduce" above

**Comment by alcpereira** [comment]
For reference, there was a fix in #5637 but it doesn't seem to work with Vite. 
Note that using `npm` it works.

**Comment by markdalgleish** [comment]
Fixed by https://github.com/remix-run/remix/pull/7904.
